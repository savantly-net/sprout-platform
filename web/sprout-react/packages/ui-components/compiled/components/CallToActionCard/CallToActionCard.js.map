{"version":3,"file":"CallToActionCard.js","sourceRoot":"","sources":["../../../src/components/CallToActionCard/CallToActionCard.tsx"],"names":[],"mappings":";AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAG1B,OAAO,EAAE,GAAG,EAAE,EAAE,EAAE,MAAM,SAAS,CAAC;AAClC,OAAO,EAAE,aAAa,EAAE,MAAM,cAAc,CAAC;AAS7C,IAAM,yBAAyB,GAAG,aAAa,CAAC,UAAC,KAAmB,IAAK,OAAA,CAAC;IACxE,OAAO,EAAE,GAAG,wRAAA,kDAEC,EAAgB,qBACb,EAAgB,wBACb,EAAsB,gHAKxC,KAPY,KAAK,CAAC,OAAO,CAAC,EAAE,EACb,KAAK,CAAC,MAAM,CAAC,GAAG,EACb,KAAK,CAAC,MAAM,CAAC,MAAM,CAAC,EAAE,CAKxC;IACD,OAAO,EAAE,GAAG,4HAAA,uBACO,EAAgB,gCAElC,KAFkB,KAAK,CAAC,OAAO,CAAC,EAAE,CAElC;IACD,MAAM,EAAE,GAAG,gGAAA,oBACK,EAAgB,OAC/B,KADe,KAAK,CAAC,OAAO,CAAC,EAAE,CAC/B;CACF,CAAC,EAlBuE,CAkBvE,CAAC,CAAC;AAEJ,MAAM,CAAC,IAAM,gBAAgB,GAAmD,UAAC,EAMhF;QALC,OAAO,aAAA,EACP,mBAAmB,yBAAA,EACnB,MAAM,YAAA,EACN,KAAK,WAAA,EACL,SAAS,eAAA;IAET,IAAM,GAAG,GAAG,yBAAyB,CAAC,KAAK,CAAC,CAAC;IAE7C,OAAO,CACL,6BAAK,SAAS,EAAE,EAAE,CAAC,CAAC,GAAG,CAAC,OAAO,EAAE,SAAS,CAAC,CAAC;QACzC,OAAO,IAAI,6BAAK,SAAS,EAAE,GAAG,CAAC,OAAO,IAAG,OAAO,CAAO;QACvD,mBAAmB;QACnB,MAAM,IAAI,6BAAK,SAAS,EAAE,GAAG,CAAC,MAAM,IAAG,MAAM,CAAO,CACjD,CACP,CAAC;AACJ,CAAC,CAAC"}