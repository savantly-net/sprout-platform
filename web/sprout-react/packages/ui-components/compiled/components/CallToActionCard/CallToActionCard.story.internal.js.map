{"version":3,"file":"CallToActionCard.story.internal.js","sourceRoot":"","sources":["../../../src/components/CallToActionCard/CallToActionCard.story.internal.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAC1B,OAAO,EAAE,wBAAwB,EAAE,MAAM,iCAAiC,CAAC;AAC3E,OAAO,EAAE,gBAAgB,EAAE,MAAM,oBAAoB,CAAC;AACtD,OAAO,EAAE,MAAM,EAAE,IAAI,EAAE,MAAM,wBAAwB,CAAC;AACtD,OAAO,EAAE,MAAM,EAAE,MAAM,kBAAkB,CAAC;AAC1C,OAAO,EAAE,MAAM,EAAE,MAAM,0BAA0B,CAAC;AAElD,eAAe;IACb,KAAK,EAAE,yBAAyB;IAChC,SAAS,EAAE,gBAAgB;CAC5B,CAAC;AAEF,MAAM,CAAC,IAAM,KAAK,GAAG;IACnB,IAAM,WAAW,GAAmC;QAClD,MAAM,EAAE,4FAAkE;QAC1E,MAAM,EAAE,CACN,oBAAC,MAAM,IAAC,IAAI,EAAC,IAAI,EAAC,IAAI,EAAC,MAAM,EAAC,OAAO,EAAE,MAAM,CAAC,oBAAoB,CAAC,qBAE1D,CACV;KACF,CAAC;IACF,IAAM,UAAU,GAAG,MAAM,CACvB,wBAAwB,EACxB;QACE,MAAM,EAAE,QAAQ;QAChB,MAAM,EAAE,QAAQ;KACjB,EACD,QAAQ,CACT,CAAC;IAEF,OAAO,wBAAwB,CAAC,gBAAgB,EAAE;QAChD,OAAO,EAAE,IAAI,CAAC,wBAAwB,EAAE,8BAA8B,CAAC;QACvE,mBAAmB,EAAE,WAAW,CAAC,UAAU,CAAC;QAC5C,MAAM,EAAE,IAAI,CAAC,uBAAuB,EAAE,6BAA6B,CAAC;KACrE,CAAC,CAAC;AACL,CAAC,CAAC"}