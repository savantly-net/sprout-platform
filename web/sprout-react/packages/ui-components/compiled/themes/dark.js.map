{"version":3,"file":"dark.js","sourceRoot":"","sources":["../../src/themes/dark.ts"],"names":[],"mappings":";AAAA,OAAO,YAAY,EAAE,EAAE,mBAAmB,EAAE,MAAM,WAAW,CAAC;AAC9D,OAAO,EAAE,gBAAgB,EAAgB,MAAM,sBAAsB,CAAC;AAEtE,IAAM,WAAW,yBACZ,mBAAmB,KACtB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,MAAM,EAAE,SAAS,EACjB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,KAAK,EAAE,SAAS,EAChB,OAAO,EAAE,SAAS,EAClB,QAAQ,EAAE,SAAS,EACnB,SAAS,EAAE,SAAS,EACpB,UAAU,EAAE,SAAS,EACrB,GAAG,EAAE,SAAS,EACd,MAAM,EAAE,SAAS,EACjB,MAAM,EAAE,SAAS,EACjB,QAAQ,EAAE,SAAS,EACnB,MAAM,EAAE,SAAS,EACjB,UAAU,EAAE,SAAS,GACtB,CAAC;AAEF,IAAM,WAAW,GAAG;IAClB,GAAG,EAAE,WAAW,CAAC,MAAM;IACvB,GAAG,EAAE,WAAW,CAAC,MAAM;IACvB,GAAG,EAAE,WAAW,CAAC,MAAM;IACvB,WAAW,EAAE,WAAW,CAAC,MAAM;IAC/B,OAAO,EAAE,WAAW,CAAC,MAAM;IAC3B,OAAO,EAAE,WAAW,CAAC,MAAM;CAC5B,CAAC;AAEF,IAAM,OAAO,GAAG;IACd,OAAO,EAAE,WAAW,CAAC,MAAM;IAC3B,OAAO,EAAE,WAAW,CAAC,MAAM;IAC3B,OAAO,EAAE,WAAW,CAAC,MAAM;CAC5B,CAAC;AAEF,IAAM,UAAU,GAAG;IACjB,UAAU,EAAE,WAAW,CAAC,MAAM;IAC9B,WAAW,EAAE,WAAW,CAAC,KAAK;IAC9B,IAAI,EAAE,WAAW,CAAC,MAAM;IACxB,YAAY,EAAE,WAAW,CAAC,MAAM;IAChC,QAAQ,EAAE,WAAW,CAAC,MAAM;IAC5B,SAAS,EAAE,WAAW,CAAC,MAAM;IAC7B,QAAQ,EAAE,WAAW,CAAC,MAAM;CAC7B,CAAC;AAEF,IAAM,IAAI,GAAG;IACX,mCAAmC;IACnC,SAAS,EAAE,UAAU,CAAC,YAAY;IAClC,eAAe,EAAE,WAAW,CAAC,MAAM;IACnC,WAAW,EAAE,WAAW,CAAC,MAAM;IAC/B,mBAAmB,EAAE,WAAW,CAAC,MAAM;IACvC,eAAe,EAAE,OAAO,CAAC,OAAO;IAChC,oBAAoB,EAAE,WAAW,CAAC,MAAM;IACxC,qBAAqB,EAAE,WAAW,CAAC,MAAM;IACzC,sBAAsB,EAAE,WAAW,CAAC,KAAK;IACzC,wBAAwB,EAAE,UAAU,CAAC,SAAS;IAC9C,aAAa,EAAE,WAAW,CAAC,MAAM;IACjC,qBAAqB,EAAE,WAAW,CAAC,MAAM;IACzC,gBAAgB,EAAE,WAAW,CAAC,MAAM;IACpC,yBAAyB,EAAE,WAAW,CAAC,KAAK;IAC5C,uBAAuB,EAAE,WAAW,CAAC,KAAK;IAC1C,YAAY,EAAE,WAAW,CAAC,MAAM;IAChC,kBAAkB,EAAE,WAAW,CAAC,MAAM;IACtC,iBAAiB,EAAE,WAAW,CAAC,MAAM;IACrC,uBAAuB,EAAE,WAAW,CAAC,MAAM;IAC3C,oBAAoB,EAAE,WAAW,CAAC,MAAM;IACxC,aAAa,EAAE,WAAW,CAAC,MAAM;IACjC,qBAAqB,EAAE,WAAW,CAAC,MAAM;IACzC,0BAA0B,EAAE,WAAW,CAAC,MAAM;IAC9C,qBAAqB,EAAE,WAAW,CAAC,MAAM;CAC1C,CAAC;AAEF,IAAM,SAAS,yBACV,YAAY,KACf,IAAI,EAAE,gBAAgB,CAAC,IAAI,EAC3B,MAAM,EAAE,IAAI,EACZ,OAAO,EAAE,KAAK,EACd,IAAI,EAAE,cAAc,EACpB,OAAO,wBACF,WAAW,KACd,YAAY,EAAE,WAAW,CAAC,MAAM,EAChC,YAAY,EAAE,WAAW,CAAC,SAAS,EACnC,YAAY,EAAE,WAAW,CAAC,MAAM,EAChC,WAAW,EAAE,WAAW,CAAC,OAAO,EAChC,QAAQ,EAAE,WAAW,CAAC,OAAO,EAC7B,UAAU,EAAE,SAAS,EACrB,WAAW,EAAE,SAAS,EACtB,WAAW,EAAE,WAAW,CAAC,MAAM,EAC/B,MAAM,EAAE,WAAW,CAAC,SAAS,EAC7B,IAAI,EAAE,SAAS,EACf,QAAQ,EAAE,WAAW,CAAC,OAAO,KAE/B,MAAM,mDACD,WAAW,GACX,OAAO,GACP,IAAI,GACJ,UAAU,KAEb,MAAM,EAAE,WAAW,CAAC,GAAG,EACvB,OAAO,EAAE,WAAW,CAAC,GAAG,EACxB,YAAY,EAAE,WAAW,CAAC,GAAG,EAC7B,gBAAgB,EAAE,OAAO,CAAC,OAAO,EACjC,WAAW,EAAE,OAAO,CAAC,OAAO,EAE5B,UAAU,EAAE,IAAI,CAAC,WAAW,EAC5B,cAAc,EAAE,WAAW,CAAC,KAAK,EACjC,qBAAqB,EAAE,WAAW,CAAC,GAAG,EAEtC,IAAI,EAAE,WAAW,CAAC,KAAK,EACvB,YAAY,EAAE,WAAW,CAAC,KAAK,EAC/B,SAAS,EAAE,WAAW,CAAC,KAAK,EAC5B,YAAY,EAAE,WAAW,CAAC,MAAM,KAElC,OAAO,EAAE;QACP,QAAQ,EAAE,MAAM;KACjB,GACF,CAAC;AAEF,eAAe,SAAS,CAAC"}