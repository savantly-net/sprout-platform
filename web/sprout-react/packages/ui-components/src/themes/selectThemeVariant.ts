import { GrafanaThemeType } from '@grafana/data';

type VariantDescriptor = { [key in GrafanaThemeType]: string | number };

export const selectThemeVariant = (variants: VariantDescriptor, currentTheme?: GrafanaThemeType) => {
  return variants[currentTheme || GrafanaThemeType.Dark];
};
