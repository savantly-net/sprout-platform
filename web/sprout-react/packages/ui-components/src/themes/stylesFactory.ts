import memoizeOne from 'memoize-one';
// import { KeyValue } from '@savantly/sprout-api';

/**
 *  Creates memoized version of styles creator
 * @param stylesCreator function accepting dependencies based on which styles are created
 */
export function stylesFactory<ResultFn extends (this: any, ...newArgs: any[]) => ReturnType<ResultFn>>(
  stylesCreator: ResultFn
) {
  return memoizeOne(stylesCreator);
}
