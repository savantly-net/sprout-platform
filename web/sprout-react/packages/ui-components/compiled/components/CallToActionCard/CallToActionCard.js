import { __makeTemplateObject } from "tslib";
import React from 'react';
import { css, cx } from 'emotion';
import { stylesFactory } from '../../themes';
var getCallToActionCardStyles = stylesFactory(function (theme) { return ({
    wrapper: css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    label: call-to-action-card;\n    padding: ", ";\n    background: ", ";\n    border-radius: ", ";\n    display: flex;\n    flex-direction: column;\n    align-items: center;\n    justify-content: center;\n  "], ["\n    label: call-to-action-card;\n    padding: ", ";\n    background: ", ";\n    border-radius: ", ";\n    display: flex;\n    flex-direction: column;\n    align-items: center;\n    justify-content: center;\n  "])), theme.spacing.lg, theme.colors.bg2, theme.border.radius.md),
    message: css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    margin-bottom: ", ";\n    font-style: italic;\n  "], ["\n    margin-bottom: ", ";\n    font-style: italic;\n  "])), theme.spacing.lg),
    footer: css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n    margin-top: ", ";\n  "], ["\n    margin-top: ", ";\n  "])), theme.spacing.lg),
}); });
export var CallToActionCard = function (_a) {
    var message = _a.message, callToActionElement = _a.callToActionElement, footer = _a.footer, theme = _a.theme, className = _a.className;
    var css = getCallToActionCardStyles(theme);
    return (React.createElement("div", { className: cx([css.wrapper, className]) },
        message && React.createElement("div", { className: css.message }, message),
        callToActionElement,
        footer && React.createElement("div", { className: css.footer }, footer)));
};
var templateObject_1, templateObject_2, templateObject_3;
//# sourceMappingURL=CallToActionCard.js.map