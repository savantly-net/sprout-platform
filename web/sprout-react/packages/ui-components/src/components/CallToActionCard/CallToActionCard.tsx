import React from 'react';
import { Themeable } from '../../types/theme';
import { GrafanaTheme } from '@grafana/data';
import { css, cx } from 'emotion';
import { stylesFactory } from '../../themes';

export interface CallToActionCardProps extends Themeable {
  message?: string | JSX.Element;
  callToActionElement: JSX.Element;
  footer?: string | JSX.Element;
  className?: string;
}

const getCallToActionCardStyles = stylesFactory((theme: GrafanaTheme) => ({
  wrapper: css`
    label: call-to-action-card;
    padding: ${theme.spacing.lg};
    background: ${theme.colors.bg2};
    border-radius: ${theme.border.radius.md};
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  `,
  message: css`
    margin-bottom: ${theme.spacing.lg};
    font-style: italic;
  `,
  footer: css`
    margin-top: ${theme.spacing.lg};
  `,
}));

export const CallToActionCard: React.FunctionComponent<CallToActionCardProps> = ({
  message,
  callToActionElement,
  footer,
  theme,
  className,
}) => {
  const css = getCallToActionCardStyles(theme);

  return (
    <div className={cx([css.wrapper, className])}>
      {message && <div className={css.message}>{message}</div>}
      {callToActionElement}
      {footer && <div className={css.footer}>{footer}</div>}
    </div>
  );
};
