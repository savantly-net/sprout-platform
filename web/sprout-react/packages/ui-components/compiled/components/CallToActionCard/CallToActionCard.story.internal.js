import React from 'react';
import { renderComponentWithTheme } from '../../utils/storybook/withTheme';
import { CallToActionCard } from './CallToActionCard';
import { select, text } from '@storybook/addon-knobs';
import { Button } from '../Button/Button';
import { action } from '@storybook/addon-actions';
export default {
    title: 'Layout/CallToActionCard',
    component: CallToActionCard,
};
export var basic = function () {
    var ctaElements = {
        custom: React.createElement("h1", null, "This is just H1 tag, you can any component as CTA element"),
        button: (React.createElement(Button, { size: "lg", icon: "plus", onClick: action('cta button clicked') }, "Add datasource")),
    };
    var ctaElement = select('Call to action element', {
        Custom: 'custom',
        Button: 'button',
    }, 'custom');
    return renderComponentWithTheme(CallToActionCard, {
        message: text('Call to action message', 'Renders message prop content'),
        callToActionElement: ctaElements[ctaElement],
        footer: text('Call to action footer', 'Renders footer prop content'),
    });
};
//# sourceMappingURL=CallToActionCard.story.internal.js.map