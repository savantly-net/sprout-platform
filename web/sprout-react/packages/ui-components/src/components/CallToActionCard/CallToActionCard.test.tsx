import React, { useContext } from 'react';
import { render } from 'enzyme';
import { CallToActionCard, CallToActionCardProps } from './CallToActionCard';
import { ThemeContext } from '../../themes';

type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

const TestRenderer = (props: Omit<CallToActionCardProps, 'theme'>) => {
  const theme = useContext(ThemeContext);
  return <CallToActionCard theme={theme} {...props} />;
};

describe('CallToActionCard', () => {
  describe('rendering', () => {
    it('when no message and footer provided', () => {
      const tree = render(<TestRenderer callToActionElement={<a href="http://dummy.link">Click me</a>} />);
      expect(tree).toMatchSnapshot();
    });

    it('when message and no footer provided', () => {
      const tree = render(
        <TestRenderer message="Click button bellow" callToActionElement={<a href="http://dummy.link">Click me</a>} />
      );
      expect(tree).toMatchSnapshot();
    });

    it('when message and footer provided', () => {
      const tree = render(
        <TestRenderer
          message="Click button bellow"
          footer="footer content"
          callToActionElement={<a href="http://dummy.link">Click me</a>}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
