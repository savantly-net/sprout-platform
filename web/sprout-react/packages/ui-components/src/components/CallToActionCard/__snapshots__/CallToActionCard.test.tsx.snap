// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToActionCard rendering when message and footer provided 1`] = `
<div
  class="css-ujo8b3-call-to-action-card"
>
  <div
    class="css-m2iibx"
  >
    Click button bellow
  </div>
  <a
    href="http://dummy.link"
  >
    Click me
  </a>
  <div
    class="css-1sg2huk"
  >
    footer content
  </div>
</div>
`;

exports[`CallToActionCard rendering when message and no footer provided 1`] = `
<div
  class="css-ujo8b3-call-to-action-card"
>
  <div
    class="css-m2iibx"
  >
    Click button bellow
  </div>
  <a
    href="http://dummy.link"
  >
    Click me
  </a>
</div>
`;

exports[`CallToActionCard rendering when no message and footer provided 1`] = `
<div
  class="css-ujo8b3-call-to-action-card"
>
  <a
    href="http://dummy.link"
  >
    Click me
  </a>
</div>
`;
